### PR TITLE
increase rclone default job deadline and make it configurable

### LINKED ISSUE #2173: [1] Rclone jobs stuck in running as their pods stop after some time
### Experienced behaviour

Discovered that rclone jobs remain stuck in running and their associated pods are stopped after about 10 minutes.
When rerunning the jobs and looking at the logs, for some of them I got the error just before the pod is terminated:
```
INFO  : Signal received: terminated
ERROR : indices/iGoJDsmKQxiyk42HiYIZ8w/0/__NT8ET9A4Tvu3LD3KRZ2T0w: Failed to copy: multipart upload failed to upload part: RequestCanceled: request context canceled
caused by: context canceled
INFO  : Exiting...
```
On some got less:
```
INFO  : Signal received: terminated
INFO  : Exiting...
```
And most of the pods just exited with nothing in the logs.
After that the jobs are stuck in running.
This was observed in many different environments on different Infra Providers all running variations of apps v0.38 including the branch [v0.38.1+rclone-netpol-fix+harbor-alert-fix](https://github.com/elastisys/compliantkubernetes-apps/compare/v0.38.1+rclone-netpol-fix+harbor-alert-fix)

### Expected behaviour

The rclone job runs without issues till completion

### Steps to reproduce

Create a  rclone job from existing cronjobs (a good candidate would be opensearch cronjob)
If the job runs for more than 10 min's it will fail as described above.

### Compliant Kubernetes Apps Version

v0.38.*

### Kubernetes Version

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Eliastisys**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

- increases default rclone job deadline from 600 to 7200
- makes ``activeDeadlineSeconds`` an optional config in sc-config:

```yaml
objectStorage:
  sync:
    activeDeadlineSeconds: 1337
```

- Fixes #2173 

#### Information to reviewers

Not sure if changes to schema are correct

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [x] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [x] The change updates the config *and* the schema
- Metrics checks:
  - [x] The metrics are still exposed and present in Grafana after the change
  - [x] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [x] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [x] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [x] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

**Review by raviranjanelastisys — APPROVED** [review]

File: config/sc-config.yaml [hunk 1]
```diff
@@ -60,6 +60,8 @@ objectStorage:
 
   ## Off-site backup replication between two providers or regions using rclone sync
   sync:
+
+    activeDeadlineSeconds: 7200
```
Comment (simonklb) [inline config/sc-config.yaml:64]:
```suggestion
    activeDeadlineSeconds: 7200
```

File: config/schemas/config.yaml [hunk 2]
```diff
@@ -1962,6 +1962,10 @@ properties:
           Sync object storage from the primary site to a secondary site with Rclone.
         type: object
         properties:
+          activeDeadlineSeconds:
+            title: Rclone cronjob activeDeadlineSeconds
```
Comment (Zash) [inline config/schemas/config.yaml:1966]: Schema LGTM. Possible title improvement:

```suggestion
            title: Rclone Cronjob Active Deadline Seconds
```

**Review by Zash — APPROVED** [review]

File: config/schemas/config.yaml (hunk 2 above)
Comment (simonklb) [reply config/schemas/config.yaml:1966]: If I had no idea what active deadline seconds was this wouldn't help much. What about something like `The maximum amount of time that the Rclone job is allowed to run (in seconds).`?

**Review by simonklb — APPROVED** [review]

File: config/schemas/config.yaml (hunk 2 above)
Comment (simonklb) [reply config/schemas/config.yaml:1966]: Ah sorry, this was the title, not the description. Ignore my previous comment. :smile:

Comment (Zash) [reply config/schemas/config.yaml]: A `description` with a link to <https://kubernetes.io/docs/concepts/workloads/controllers/job/> perhaps?

**Pull request merged**